### PR TITLE
run manager containers

### PR DESCRIPTION
**Pull request by bloodearnest**
run the management containers for the duration, and add labels to management containers and and volumes

### CODE REVIEW

File: jobrunner/lib/docker.py [hunk 1]
```diff
@@ -47,6 +47,13 @@ class DockerDiskSpaceError(Exception):
     pass
 
 
+def docker_labels(job):
+    return [
+        "--label", f"action={job.action}",
+        "--label", f"workspace={job.workspace}",
+    ]
```
Comment (evansd) [inline jobrunner/lib/docker.py:54]: Semantic pedantry alert: I'm aware I've not written down my mental model of this anywhere, but to me this is the wrong place for a function like this. I think of `lib/docker.py` as a generic "make docker easier to interact" with module, so not one that knows about "jobs" as such, or other application level concerns. So I'd rather be passing the labels in and have the function which generates them somewhere else.

**Review by evansd — CHANGES_REQUESTED** [review]
Thanks, I think this looks like a sensible way of doing things. A couple of pedantic comments and a substantive one about the error handling when removing a container.

File: jobrunner/lib/docker.py [hunk 2]
```diff
@@ -126,13 +126,17 @@ def delete_volume(volume_name):
...
+            ["rm", "--force", manager_name(volume_name)],
             check=True,
             capture_output=True,
         )
     except subprocess.CalledProcessError as e:
         # Ignore error if container has already been removed
-        if e.returncode != 1 or b"No such container" not in e.stderr:
+        if (
+            e.returncode != 1
+            or b"Cannot kill container" not in e.stderr
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply jobrunner/lib/docker.py:137]: This is a bit of a left over from trying to be clever and do `docker run --rm` then `docker kill`, which caused issues.

But I left the case in, as I thought `docker rm --force` used SIGTERM, which could in theory fail and produce that message. I've realised now that it uses SIGKILL, so if that fails we've got bigger problems. Will remove.

File: jobrunner/lib/docker.py [hunk 3]
```diff
@@ -70,43 +77,36 @@ def docker(docker_args, timeout=DEFAULT_TIMEOUT, **kwargs):
...
-            ],
-            check=True,
-            capture_output=True,
+        run(
+            manager_name(volume_name),
+            [MANAGEMENT_CONTAINER_IMAGE, "sh"],
+            volume=(volume_name, VOLUME_MOUNT_POINT),
+            label=LABEL,
+            job=job,
+            allow_network_access=True,
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply jobrunner/lib/docker.py:102]: Tests pass without it, so have removed.

File: jobrunner/lib/docker.py (hunk 1 above)
Comment (bloodearnest) [reply jobrunner/lib/docker.py]: Have addressed, good call. I felt icky writing it, just couldn't quite get to how to do it better.

**Review by evansd — APPROVED** [review]

**Pull request merged**